Code/Framework/AzFramework/Tests/Spawnable/SpawnableScriptMediatorTests.cpp

```diff
@@ -0,0 +1,302 @@
+            specializations.Append("test");
+            specializations.Append("spawnable");
+        }
+    };
+    
+    class SpawnableScriptMediatorTests
+        : public AllocatorsFixture
+        , public AzFramework::Scripts::SpawnableScriptNotificationsBus::MultiHandler
+    {
+    public:
+        using TicketToEntitiesPair = AZStd::pair<AzFramework::EntitySpawnTicket, AZStd::vector<AZ::EntityId>>;
+        constexpr static AZ::u64 EntityIdStartId = 40;
+
+        AZStd::vector<TicketToEntitiesPair> m_spawnedEntities;
+
+        void OnSpawn(AzFramework::EntitySpawnTicket spawnTicket, AZStd::vector<AZ::EntityId> entityList) override
+        {
+            m_spawnedEntities.push_back({ spawnTicket, AZStd::move(entityList) });
+        }
+
+        virtual void OnDespawn(AzFramework::EntitySpawnTicket spawnTicket) override
+        {
+            AZStd::erase_if(
+                m_spawnedEntities,
+                [&spawnTicket](const TicketToEntitiesPair& pair) -> bool
+                {
+                    return pair.first == spawnTicket;
+                });
+        }
+
+        void SetUp() override
+        {
+            AllocatorsFixture::SetUp();
+
+            m_application = new TestApplication();
+            AZ::ComponentApplication::Descriptor descriptor;
+            m_application->Start(descriptor);
+
+            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
+            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash
+            // in the unit tests.
+            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+
+            auto managerInterface = AzFramework::SpawnableEntitiesInterface::Get();
+            m_manager = azrtti_cast<AzFramework::SpawnableEntitiesManager*>(managerInterface);
+        }
+
+        void TearDown() override
+        {
+            AzFramework::Scripts::SpawnableScriptNotificationsBus::MultiHandler::BusDisconnect();
+            // One more tick on the spawnable entities manager in order to delete the ticket fully.
+            while (m_manager->ProcessQueue(
+                       AzFramework::SpawnableEntitiesManager::CommandQueuePriority::High |
+                       AzFramework::SpawnableEntitiesManager::CommandQueuePriority::Regular) !=
+                   AzFramework::SpawnableEntitiesManager::CommandQueueStatus::NoCommandsLeft)
+                ;
+            
+            m_spawnedEntities = {};
+
+            delete m_application;
+            m_application = nullptr;
+
+            AllocatorsFixture::TearDown();
+        }
+
+        void WaitForResponse(AzFramework::Scripts::SpawnableScriptMediator& mediator)
+        {
+            for (size_t i=0; i<1000; ++i) // Don't do this indefinitely to avoid deadlocking on a failing test.
```

Comment:
If you are waiting for a response from another thread, you should do some sort of thread::sleep inside the loop

Reply:
Excellent!